yt-dlp-py.py

```diff
@@ -1,53 +1,65 @@
 import subprocess
 from pprint import pprint
 import inquirer
-typeQuestion = [
-    inquirer.List(
-        "type",
-        message="What type of import are you using?",
-        choices=["batch", "link"],
-    ),
-]
-typeState = inquirer.prompt(typeQuestion)
+from classes import FileFormat, ImportType, UploadInformation
+
+#region Helper Functions
+def promptForFormat():
+    return [inquirer.List(
+            UploadInformation.FORMAT,
+            message="What format do you need?",
+            choices=[FileFormat.MP4, FileFormat.MP3, FileFormat.WEBM],
+            )]
+def promptForImportType():
+    return [inquirer.List(
+            UploadInformation.TYPE,
+            message="What type of import are you using?",
+            choices=[ImportType.BATCH, ImportType.LINK],
+            )]
+    
+def batchUploadVideo(batchFilePath, format):
+    subprocess.run(["./Assets/yt-dlp", "-f", format, "--ffmpeg-location", "./Assets/ffmpeg-2024-10-02-git-358fdf3083-full_build/bin/ffmpeg.exe", "-P","Output", "-a", batchFilePath])
+
+def batchUploadSound(batchFilePath):
+    subprocess.run(["./Assets/yt-dlp", "-x","--audio-format","mp3","--ffmpeg-location", "./Assets/ffmpeg-2024-10-02-git-358fdf3083-full_build/bin/ffmpeg.exe", "-P","Output", "-a", batchFilePath])
+
+def linkUploadVideo(youTubeLink, format):
+    subprocess.run(["./Assets/yt-dlp", "-f", format, "--ffmpeg-location", "./Assets/ffmpeg-2024-10-02-git-358fdf3083-full_build/bin/ffmpeg.exe", "-P","Output", youTubeLink])
+
+def linkUploadSound(youTubeLink):
+    subprocess.run(["./Assets/yt-dlp", "-x","--audio-format","mp3","--ffmpeg-location", "./Assets/ffmpeg-2024-10-02-git-358fdf3083-full_build/bin/ffmpeg.exe", "-P","Output", youTubeLink])
+
+def uploadMedia(format, media, videoUploadFunc, soundUploadFunc):
+    match format:
+        case FileFormat.MP4:
+            videoUploadFunc(media, FileFormat.MP4)
+        case FileFormat.WEBM:
+            videoUploadFunc(media, FileFormat.WEBM)
+        case FileFormat.MP3:
+            soundUploadFunc(media)
+        case _:
+            print("Please select valid download option and try again")
+#endregion
+
+#region Main
+# Prompt user for import type
+typeState = inquirer.prompt(promptForImportType())
 pprint(typeState)
-if typeState["type"] == "batch":
+match typeState[UploadInformation.TYPE]:
```

Comment:
One thing I stride to do is reduce nesting. Nesting is when several decision points are within each other. A common example is below:

```
if (thisIsTrue) {
  if (otherThingIsTrue) {
    doOtherThing()
  } else {
    doFirstThing()
  }
} else {
  doThirdThing()
}
```

This can be simplified too: 

```
if (!thisIsTrue) {
  doThirdThing()
  return
}
if (otherThingIsTrue) {
  doOtherThing()
  return;
} 
doFirstThing()
```

Now for this PR I reduced nesting by using methods rather than short circuits